main.go

```diff
@@ -381,19 +426,54 @@ func runCLIApp(c *cli.Context) (err error) {
 	{
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
-		if err == nil {
+		// This utility is to absorb the error
+		// returned by daemonize.SignalOutcome calls by simply
+		// logging them as error logs.
+		callDaemonizeSignalOutcome := func(err error) {
+			if err2 := daemonize.SignalOutcome(err); err2 != nil {
+				logger.Errorf("Failed to signal to daemon: %v", err2)
+			}
+		}
+
+		markSuccessfulMount := func() {
 			// Print the success message in the log-file/stdout depending on what the logger is set to.
 			logger.Info(SuccessfulMountMessage)
+			callDaemonizeSignalOutcome(nil)
+		}
 
-			daemonize.SignalOutcome(nil)
-		} else {
+		markFailure := func(err error) {
 			// Printing via mountStatus will have duplicate logs on the console while
 			// mounting gcsfuse in foreground mode. But this is important to avoid
 			// losing error logs when run in the background mode.
 			logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
 			err = fmt.Errorf("%s: mountWithArgs: %w", UnsuccessfulMountMessagePrefix, err)
-			daemonize.SignalOutcome(err)
-			return
+			callDaemonizeSignalOutcome(err)
+		}
+
+		if err != nil {
+			markFailure(err)
+			return err
+		}
+		if isDynamicMount {
+			markSuccessfulMount()
+		} else {
+			switch flags.MetadataPrefetchOnMount {
```

Comment:
We can log the SuccessfulMountMessage once at the end and save three lines of code.
The goroutine shouldn't really matter since anyway we are not waiting for the goroutine to finish to log the success message.